### ISSUE TITLE
How to check if two components overlap or collide in PDFViewer? 

### ISSUE THREAD

**Issue by handao-everfit** [diegomura/react-pdf#1519]
Is it possible to determine whether two View components overlap or collide with each other. I'm trying to check whether if a component touches the page footer so that I could break the component into two parts and display them on two different pages?

I've tried to using getBoundingClientRect() to check for collision but the function returns DOMRect {x: 0, y: 0, width: 0, height: 0, top: 0, …} for components contained inside PDFViewer.

![image](https://user-images.githubusercontent.com/87088870/134441799-94cfefbb-7f4f-4be8-83ce-41c88e1d9138.png)

Browser: Chrome 93
@react-pdf/renderer version: 2.0.19
React version: 17.0.2

**Comment by diegomura** [comment]
I'm not sure if I understood You mean two components inside the document? I see you have a div inside the PDFViewer which is not really supported

**Comment by handao-everfit** [comment]
Yes, two View components inside the Document. I just tried to see if it works with a div. I'm trying to check whether if a View component touches the page footer or overflows to the next page so that I could break the View component into two smaller parts and display them on two different pages.

**Comment by handao-everfit** [comment]
> I'm not sure if I understood You mean two components inside the document? I see you have a div inside the PDFViewer which is not really supported

Yes, two View components inside the Document. I just tried to see if it works with a div. I'm trying to check whether if a View component touches the page footer or overflows to the next page so that I could break the View component into two smaller parts and display them on two different pages.

**Comment by diegomura** [comment]
Not possible right now. Flagging it as new feature, although I'm not sure if will be supported

**Comment by handao-everfit** [comment]
> Not possible right now. Flagging it as new feature, although I'm not sure if will be supported

Thank you! Is there a way to know whether if a component breaks then?
